### DISCUSSION TITLE
Registry IP Strategy

### DISCUSSION THREAD

**Discussion by esoterra** [bytecodealliance/SIG-Registries#37] [General]
One risk the registry project faces is that some 3rd party tries to claim key IP needed to implement the registry project. They could use this to prevent others from implementing it. We need to decide on an approach to prevent this.

As we continue to write, license, and publish code and content describing the registry, we will create a collection of prior art that can be used to prevent others from acquiring and/or enforcing patents against the registry project.

Additionally, we could pursue patents on key registry IP, which will only ever be used defensively, to prevent others from attempting to license/monopolize registry IP. This would put us "already in the system" so that patent examiners evaluating a new patent against registry technology could easily find and refer to our existing patent(s).

Should we pursue patent(s) to defend registry IP as open and accessible or rely on copyright law and published prior art?

**Comment by ricochet** [comment]
@squillace is going to raise this with the board

**Reply by esoterra** [reply]
@squillace can you also ask about whether we should trademark or otherwise protect "warg" from being sniped/controlled. We're interested in using it at as a URL scheme to refer to packages in the registry e.g. `warg://bytecodealliance.org/foobar/1.0`

**Comment by squillace** [comment]
Some responses emerge:

1. Basically, up-front work to establish prior art is typically a wasteful use of energy when prior art is established by publishing in the first place. 
2. Doing the up front work is for offensive capabilities, and the BA has no interest in offense or investments in them. 
3. Defense against possible "competition" (???) is not something that the BCA has in its mission. In fact, we welcome the usage of all our work.
4. The current BCA IP policy includes the Apache patent grant without any requirement that contributors assign IP. 
5. Finally, to partially restate point 3 slightly differently, it's just not compatible with our sense of the open source spirit. We exist to build and share.

As a result, the Board's take on the discussion is no, we're fine the way we are, BUT with one recommendation: _publish early and often_. Establishing public prior art is the way the BCA can defend the work done properly so that everyone can use it without encumbrance of any sort. 

Does this help guide you all?

**Reply by esoterra** [reply]
Thanks Ralph, I appreciate the guidance. The clarification that patents don't have a place in the foundation's OSS philosophy is really helpful. This question was simply due-diligence to say that we had asked the question of whether freely licensed patents have a place defending open source software from attempts by actors to privately/proprietorially license it.

We're going to go public with more information for Wasm Day with a presentation, blog post, and code to get started publishing as early and often as possible. Thanks for your help!

**Comment by squillace** [comment]
As to `warg`, typically the way I've understood you grab that name is by grabbing the domain and registering the prefix for the URI scheme with iana. https://en.wikipedia.org/wiki/List_of_URI_schemes. That's as much of a public statement as anything else.

As a corp guy, however, you def need to ensure warg is in the public domain; it is, I believe, but due diligence and all that. Changing protocol schemes is a pain.

**Reply by kulakowski-wasm** [reply]
https://github.com/bytecodealliance/SIG-Registries/discussions/39#discussioncomment-3789621 yup! I think the only thing in the way now is the paperwork, and in particular deciding exactly who/what is the point of contact.

**Reply by squillace** [reply]
it's possibly best a BCA registration, not sure. This is all new to us as a foundation, so I'll ask.

**Comment by squillace** [comment]
interestingly, in the US, the only warg TM is for a line of clothing, so no TM collisions are extant. There is https://www.wargtechnologies.com/ but they seem to be mainly a small local startup in India. Nothing in GH pops up, so that's reasonably clear.
